### ISSUE TITLE
How to identify CNN learned motifs

### ISSUE THREAD

**Issue by JiaqiLiZju** [p-koo/learning_sequence_motifs#2]
Nice work! And the benchmark really helps!

When I try to reproduce your work, I've met the important problem that `The large discrepancy between the number of filters that match any motif in the JASPAR database versus the number that match ground truth motifs demonstrates that motif comparison is not a reliable tool to identify whether a CNN learns relevant motifs.` in PLOS Computational Biology | https://doi.org/10.1371/journal.pcbi.1007560

Here are some questions:
1. Even if I got the ground truth, I found it's difficult to identify those CNN learned motifs. But I wonder whether this problem caused by the feature-map based methods used in DeepOmics? activated feature map in a specific channel may mess up two distinct motifs, I think. 

2. It will be grateful for me if you could recommend some useful tools to identify CNN learned motifs, especially when I have no  ground truth.

**Comment by p-koo** [comment]
For motif discovery, I recommend using a battery of approaches: 
1. visualize filters -- it's important to use a network that has a strong inductive bias to learn motifs in 1st layer filters. Recently, we published this trick that seems to really help our networks -- exponential activations in 1st layer only (https://www.nature.com/articles/s42256-020-00291-x). Note, batch norm is key to making this trick work.
2. feature attribution methods -- saliency maps, smoothgrad, integrated gradients, deeplift, or deepshap, or mutagenesis. They can vary at times, but if the model is good, all give similar-ish results. You can look at the above paper for a comparison. We find that exp activations also helps improve efficacy of attribution methods. There are other tricks that can achieve similar results, such as attribution priors, but they may require more sophisticated training.
3. global importance analysis -- while you can identify putative motifs, it's not always clear whether you have the right one. You can run in silico experiments, but designed like a randomized control trial. GIA provides this framework and you can find more info here: https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008925. The kmer discovery experiment allows you to de novo discover top k-mers and you can use standard bioinformatics tools to assemble a more standard motif-like representation.

**Comment by JiaqiLi1024** [comment]
Thanks a lot!
